### DISCUSSION TITLE
How to use WebMidi in a JS Worker

### DISCUSSION THREAD

**Discussion by nathanpaletat** [djipco/webmidi#291] [Q&A]
Hello, I have to use webmidi.js inside a javascript worker.

I try with importScripts("webmidi.iife.min.js") inside my worker file but I can't get 'WebMidi' object...

I don't find the way... please help me !
Thanks a lot

**Comment by nathanpaletat** [comment]
If I try

import {WebMidi} from "https://cdn.jsdelivr.net/npm/webmidi@next/dist/esm/webmidi.esm.min.js"; inside my worker file

console log : Uncaught SyntaxError: Cannot use import statement outside a module

**Comment by djipco** [comment]
As far as I know, the Web MIDI API is not supported in workers. You can check out [this thread](https://github.com/WebAudio/web-midi-api/issues/99) for more details.
